tests/unit-tests/test_inferrer.py

```diff
@@ -391,7 +480,1004 @@ def check_eval(mock_do_mini_batch_inference,
+                )
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+
+        # Test case 4: abnormal case, set RGCN model with edge features for LP, but 
+        #              not provide edge features.
+        #              This will trigger an assertion error, asking for giving edge feature
+        #              for message passing computation.
+        create_config4ef(Path(tmpdirname), 'gnn_nc.yaml', task='lp', use_ef=True)
+        args = Namespace(yaml_config_file=os.path.join(Path(tmpdirname), 'gnn_nc.yaml'),
+                            local_rank=0)
+        config = GSConfig(args)
+        gdata = GSgnnData(part_config=part_config,
+                          node_feat_field=config.node_feat_name)
+
+        model5 = create_builtin_lp_gnn_model(gdata.g, config, True)
+        inferrer5 = GSgnnLinkPredictionInferrer(model5)
+        inferrer5.setup_device(device)
+
+        infer_dataloader5 = GSgnnLinkPredictionTestDataLoader(
+            gdata,
+            target_idx=gdata.get_edge_val_set(config.train_etype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            node_feats=config.node_feat_name,
+            edge_feats=None,
+            num_negative_edges=config.num_negative_edges)
+
+        with assert_raises(AssertionError):
+            inferrer5.infer(
+                data=gdata,
+                loader=infer_dataloader5,
+                save_embed_path='/tmp/embs',
+                use_mini_batch_infer=True
+                )
+
+    # delete temporary results
+    if os.path.exists('/tmp/embs'):
+        shutil.rmtree('/tmp/embs')
+
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+def test_hgt_infer_nc4ef():
+    """ Test HGT model Node Classification inference pipeline with/without edge features.
+    
+    Because HGT encoder dose not support edge feature so far, if initialized with edge_feat_name,
+    it will trigger a Not-support assertion error.
+    """
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+
+    setup_device(0)
+    device = get_device()
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        gdata = GSgnnData(part_config=part_config)
+
+        # Test case 0: normal case, set HGT model without edge features for NC, and not provide
+        #              edge features.
+        #              Should complete inference process
+        create_config4ef(Path(tmpdirname), 'gnn_nc.yaml', encoder='hgt', use_ef=False)
```

Comment:
We can use @pytest.mark.parametrize("encoder", ["rgcn", "hgt"]) to simplify the test code.